source/math/FloatMatrix.ooc

```diff
@@ -236,6 +236,48 @@ FloatMatrix: cover {
 		this free(Owner Receiver)
 		result
 	}
+	// Thomas algorithm for tridiagonal systems
+	// https://en.wikipedia.org/wiki/Tridiagonal_matrix_algorithm
+	solveTridiagonal: func (y: This) -> This {
+		diagonal := this take() getDiagonal() take()
```

Comment:
Sure, I will try to come up with something at some point today. I can send it to you as a draft in any case.